### DISCUSSION TITLE
Docs: Generating a sitemap with `next export`

### DISCUSSION THREAD

**Discussion by alextse** [vercel/next.js#73022] [Help]
### What is the documentation issue?

Using the provided example, getting the following error with a SSG setup.

> Error: Page "/sitemap.xml/[[...__metadata_id__]]/route" is missing exported function "generateStaticParams()", which is required with "output: export" config.

### Is there any context that might help us understand?

Next.js version: 14.2.13

Is generating sitemap.xml even possible with SSG?

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap

**Comment by samcx** [comment]
@alextse Can you clarify which example you're referring to (e.g., screenshot)?

**Comment by alextse** [comment]
Hi @samcx , below is the sample I've tried.

```typescript
import type { MetadataRoute } from 'next'
 
export default function sitemap(): MetadataRoute.Sitemap {
  return [
    {
      url: 'https://acme.com',
      lastModified: new Date(),
      changeFrequency: 'yearly',
      priority: 1,
    },
    {
      url: 'https://acme.com/about',
      lastModified: new Date(),
      changeFrequency: 'monthly',
      priority: 0.8,
    },
    {
      url: 'https://acme.com/blog',
      lastModified: new Date(),
      changeFrequency: 'weekly',
      priority: 0.5,
    },
  ]
}
```

and here's my next.js config.
```typescript
/** @type {import('next').NextConfig} */
const nextConfig = {
    output: 'export'
};

export default nextConfig;

```

**Comment by samcx** [comment]
@alextse Thanks for the clarification! Taking a look <picture data-single-emoji=":frog-eyes:" title=":frog-eyes:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/frog-eyes/558eb9b0fd306c97.png" alt=":frog-eyes:" align="absmiddle"></picture>

**Comment by fosple** [comment]
Can also confirm the issue

_Edit by maintainer bot: Comment was **automatically** minimized because it was considered unhelpful. (If you think this was by mistake, let us know). Please only comment if it adds context to the issue. If you want to express that you have the same problem, use the upvote 👍 on the issue description or subscribe to the issue for updates. Thanks!_

**Comment by samcx** [comment]
@alextse This is the error I get when I try to build locally →

```
   Collecting page data  .Error: export const dynamic = "force-static"/export const revalidate not configured on route "/sitemap.xml" with "output: export". See more info here: https://nextjs.org/docs/advanced-features/static-html-export
```
   
If you add the Route Segment Config, you are able to successfully build with `next export`. I'm curious if that should even be needed though 👁️

**Reply by alextse** [reply]
Thank you @samcx 
I've tried adding `export const dynamic = "force-static";` in `layout.tsx` or `sitemap.ts` but still getting the same error.

**Reply by samcx** [reply]
@alextse Can you send us a repro so we attempt to replicate what you're seeing?

**Comment by gndelia** [comment]
@alextse were you able to find a solution? I am facing the same issue but in next 15

**Comment by Vigrond** [comment]
confirming `export const dynamic = 'force-static'` worked for me on `app/sitemap.tsx` with `output: 'export'` using the example template [referenced in this thread](#discussioncomment-11329188)

i.e.

```ts
import type { MetadataRoute } from 'next'
```
```diff
+ export const dynamic = 'force-static'
```
```ts
export default function sitemap(): MetadataRoute.Sitemap {
  return [
    {
      url: 'https://acme.com',
      lastModified: new Date(),
      changeFrequency: 'yearly',
      priority: 1,
    },
    {
      url: 'https://acme.com/about',
      lastModified: new Date(),
      changeFrequency: 'monthly',
      priority: 0.8,
    },
    {
      url: 'https://acme.com/blog',
      lastModified: new Date(),
      changeFrequency: 'weekly',
      priority: 0.5,
    },
  ]
}
```
